src/cdk-experimental/combobox/combobox.ts

```diff
@@ -6,15 +6,192 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Directive} from '@angular/core';
+export type OpenAction = 'focus' | 'click' | 'downKey' | 'toggle';
+
+export type OpenActionInput = OpenAction | OpenAction[] | string | null | undefined;
+
+import {
+  AfterContentInit,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Optional,
+  Output, ViewContainerRef
+} from '@angular/core';
+import {CdkComboboxPanel} from './combobox-panel';
+import {TemplatePortal} from '@angular/cdk/portal';
+import {
+  ConnectedPosition,
+  FlexibleConnectedPositionStrategy,
+  Overlay,
+  OverlayConfig,
+  OverlayRef
+} from '@angular/cdk/overlay';
+import {Directionality} from '@angular/cdk/bidi';
+import {BooleanInput, coerceBooleanProperty, coerceArray} from '@angular/cdk/coercion';
+
 
 @Directive({
   selector: '[cdkCombobox]',
   exportAs: 'cdkCombobox',
   host: {
-    'role': 'combobox'
+    'role': 'combobox',
+    '(click)': 'toggle()',
+    '[attr.aria-disabled]': 'disabled'
```

Comment:
The `aria-haspopup` and `aria-controls` should go here. 

I also realized just now that there's a mode of interaction where this element controls a listbox while focus remains on the combobox element and it uses `aria-activedescendant` to indicate which listbox option is active. Seems like something to consider later on